packages/protocol/contracts/stability/Reserve.sol

```diff
@@ -281,15 +290,20 @@ contract Reserve is IReserve, Ownable, Initializable, UsingRegistry, ReentrancyG
   /*
    * Internal functions
    */
+
   /**
-   * @notice Computes a tobin tax based on the reserve ratio.
-   * @return The numerator of the tobin tax amount, where the denominator is 1000.
+   * @notice Computes the ratio of current reserve balance to total stable token valuation.
+   * @return Reserve ratio in a fixed point format.
    */
-  function computeTobinTax() private view returns (FixidityLib.Fraction memory) {
+  function getReserveRatio() public view returns (uint256) {
     address sortedOraclesAddress = registry.getAddressForOrDie(SORTED_ORACLES_REGISTRY_ID);
     ISortedOracles sortedOracles = ISortedOracles(sortedOraclesAddress);
     uint256 reserveGoldBalance = getReserveGoldBalance();
     uint256 stableTokensValueInGold = 0;
+    FixidityLib.Fraction memory cgldWeight = FixidityLib.wrap(assetAllocationWeights["cGLD"]);
+    if (cgldWeight.equals(FixidityLib.wrap(0))) {
+      cgldWeight = FixidityLib.fixed1();
+    }
```

Comment:
This seems a bit dangerous.  How about requiring here that assetAllocationWeights["cGLD"]  is nonzero and making sure that it is set to 0.5 in migrations?

Reply:
This is tricky, but I think that b/c we're using unsigned ints/fixed point numbers, it can't be negative. The only case where it is 0 is when it does not have an associated weight (either not set yet or intentionally zero).

WRT setting it in migrations, if it's required to be set early on, I think that it'd make more sense to then move it to the initialize stage. I feel like it'd just be then set to 1 by default by most people. You're right that it feels dangerous, and I'm not quite sure the best approach here.

Reply:
Not sure I understand what you mean by "I feel like it'd just be then set to 1 by default by most people." The reserve cGLD weight should be the same for everybody and only modifiable via governance. 
It's for me to judge whether the initial weights should be set in migrations or in initialization stage - any thoughts @asaj?
If we make sure that the cGLD weight must be set and cannot be set to zero (or something negative), we should be able to get rid of the above if statement.

Reply:
RE cGLD weight being set, it's mainly for testing / start up. Long term the weights will exist and be updated more infrequently.

Making the cGLD weight privileged by having it be required does make sense from a stability sense, but will require more changes in how weights are handled (no thought is currently given to what the symbols for the weights are, just that they add up to 1, in addition, (@asaj), we probably need to ensure that the symbols in the input array are unique now that we have the mapping.